test/alternator/test_metrics.py

```diff
@@ -512,6 +512,49 @@ def do_test():
             test_table_s.get_item(Key={'p': pk})
         check_histogram_metric_increases('GetItem', 'operation_size_kib', metrics, do_test, [(0, bucket(17)), (1, bucket(17.1)), (1, bucket(INF))])
 
+def test_put_item_many_items_fall_into_appropriate_buckets(test_table_s, metrics):
+    def do_test():
+        pk = random_string()
+        test_table_s.put_item(Item={'p': pk, 'a': 'a' * 1 * KB, 'b': 'b' * 5 * KB})
+        test_table_s.put_item(Item={'p': pk, 'a': 'a' * 6 * KB})
+        test_table_s.put_item(Item={'p': pk, 'a': 'a' * 401 * KB})
+    check_histogram_metric_increases('PutItem', 'operation_size_kib', metrics, do_test, [(0, prev_bucket(6.1)), (2, bucket(6.1)), (2, prev_bucket(401.1)), (3, bucket(401.1)), (3, bucket(INF))])
+
+# Verify that only the new item size is counted in the histogram. The WCU is
+# calculated as the maximum of the old and new item sizes, but the histogram
+# should log only the new item size.
+def test_put_item_increases_metrics_for_new_item_size_only(test_table_s, metrics):
+    pk = random_string()
+    points = [[value, {'op': 'PutItem', 'le': le}] for value, le in [(0, prev_bucket(6.1)), (1, bucket(6.1)), (1, bucket(INF))]]
+
+    test_table_s.put_item(Item={'p': pk, 'b': 'b' * 3 * KB})
+    with check_increases_metric_exact(metrics, 'scylla_alternator_table_operation_size_kib_bucket', points):
+        test_table_s.put_item(Item={'p': pk, 'a': 'a' * 6 * KB})
+
+@pytest.mark.parametrize("force_rbw", [True, False])
+def test_delete_item_is_zero_for_nonexistent_item(dynamodb, test_table_s, metrics, force_rbw):
+    with scylla_config_temporary(dynamodb, 'alternator_force_read_before_write', str(force_rbw).lower()):
+        def do_test():
+            test_table_s.delete_item(Key={'p': random_string()})
+        check_histogram_metric_increases('DeleteItem', 'operation_size_kib', metrics, do_test, [(0, bucket(INF))])
+
+@pytest.mark.parametrize("force_rbw", [True, False])
+def test_delete_item_many_items_fall_into_appropriate_buckets(dynamodb, test_table_s, metrics, force_rbw):
+    with scylla_config_temporary(dynamodb, 'alternator_force_read_before_write', str(force_rbw).lower()):
+        if force_rbw:
+            points = [[value, {'op': 'DeleteItem', 'le': le}] for value, le in [(0, prev_bucket(24.1)), (1, bucket(24.1)), (1, prev_bucket(378.1)), (2, bucket(378.1)), (3, bucket(INF))]]
+        else:
+            points = [[0, {'op': 'DeleteItem', 'le': bucket(INF)}]]
+
+        # ~378KiB, ~24KiB, ~401KiB
+        pks = [random_string() for _ in range(3)]
```

Comment:
We discussed in a different discussion that there is an absolutely tiny chance that two random keys here will be the same, but I believe it's too tiny to even care about. The long-term solution is to do https://github.com/scylladb/scylladb/issues/9988, so let's not bother with this now.